inc/content-front.php

```diff
@@ -15,27 +15,46 @@
 	<div class="entry-content">
 		
 				<?php
+				// Get array of sticky posts.
 				$sticky = get_option( 'sticky_posts' );
-				rsort( $sticky );
-				$sticky = array_slice( $sticky, 0, 5 );
-				$the_query = new WP_Query( array( 'post__in' => $sticky, 'ignore_sticky_posts' => 1 ) );
-				while ( $the_query->have_posts() ) {
-					$the_query->the_post();
-				?>
-				
-				<div class="excerpt-post">
-					<img class="excerpt-post__fig"  src="<?php echo get_first_post_image(); ?>" width="200" >
-					<div class="excerpt-post__body">
-		                <h3><a href="<?php echo the_permalink();?>"><?php echo get_the_title();?></a></h3>
-		                <?php custom_excerpt( 20, '...' ) ?>
-					</div>
-				</div>
-				
-				<?php } wp_reset_postdata(); ?>
 
-				<?php the_content(); ?>
-				
-				<?php if ( is_active_sidebar( 'sidebar-two' ) ) : ?>
+				// If no sticky posts, skip this whole section.
+				if ( count( $sticky ) > 0 ) {
+
+					// Build WP query.
+					$args = array(
+						'posts_per_page' => 5,
+						'post__in' => $sticky,
+						'ignore_sticky_posts' => 1,
+					);
+
+					// Execute query.
+					$the_query = new WP_Query( $args );
+
+					if ( $the_query->have_posts() ) {
+
+						while ( $the_query->have_posts() ) :
+							$the_query->the_post();
+						?>
+
+						<div class="excerpt-post">
+							<img class="excerpt-post__fig" src="<?php echo esc_attr( get_first_post_image() ); ?>" width="200" >
+							<div class="excerpt-post__body">
+				                <h3><a href="<?php echo the_permalink();?>"><?php echo get_the_title();?></a></h3>
+				                <?php custom_excerpt( 20, '...' ) ?>
+							</div>
+						</div>
+
+						<?php endwhile;
+
+						wp_reset_postdata();
+
+					} // End "have posts".
+				} // End "if no sticky posts".
+
+				the_content();
+
+				if ( is_active_sidebar( 'sidebar-two' ) ) : ?>
```

Comment:
Looks good to me!

2016-06-29 14:15 GMT-04:00 frances botsford <notifications@github.com>:

> In inc/content-front.php
> <https://github.com/MITLibraries/MITLibraries-child/pull/59#discussion_r68999355>
> :
>
> > +							<div class="excerpt-post__body">
> > +				                <h3><a href="<?php echo the_permalink();?>"><?php echo get_the_title();?></a></h3>
> > +				                <?php custom_excerpt( 20, '...' ) ?>
> > +							</div>
> > +						</div>
> > +
> > +						<?php endwhile;
> > +
> > +						wp_reset_postdata();
> > +
> > +					} // End "have posts".
> > +				} // End "if no sticky posts".
> > +
> > +				the_content();
> > +
> > +				if ( is_active_sidebar( 'sidebar-two' ) ) : ?>
>
> Ah I see now. And I agree with your cleanup ideas!
>
> —
> You are receiving this because you were mentioned.
> Reply to this email directly, view it on GitHub
> <https://github.com/MITLibraries/MITLibraries-child/pull/59/files/991cea9f73635d15db20466cb0031cc94e09d2ac#r68999355>,
> or mute the thread
> <https://github.com/notifications/unsubscribe/ABv1VYwaNXvF8GtM9XvtzaAv5DeQiei7ks5qQrZfgaJpZM4I8CLt>
> .
>



-- 

"The reason is never the reason."


*- Some French Movie Director*"Every portrait that is painted with feeling
is a portrait of the artist, not the sitter."
*- Oscar Wilde*